petastorm/etl/dataset_metadata.py

```diff
@@ -77,7 +76,10 @@ def materialize_dataset(spark, dataset_url, schema, row_group_size_mb=None):
         validate_schema=False)
 
     _generate_unischema_metadata(dataset, schema)
-    if not dataset.metadata_path:
+    try:
+        # Try to load the row groups, if it fails that means the metadata was not generated properly
+        load_row_groups(dataset)
+    except ValueError:
```

Comment:
A little strange that you catch `ValueError` here, as it indicates invalid arguments to the called function. I think we are trying to communicate something more concrete. Maybe a custom error class for this case is more appropriate?